### PR TITLE
Support JDK 11 and JDK 17 Compilation

### PR DESCRIPTION
**Pull request by YogyaGamage**
### Purpose
Add support for JDK 11 and JDK 17 compilation.

### When should this PR be merged
After this PR is merged to the main branch, we can't build the branch on JDK 8

### Related Issue
https://github.com/wso2/product-is/issues/14224

### CODE REVIEW

File: components/org.wso2.carbon.identity.oauth.dcr.endpoint/pom.xml [hunk 1]
```diff
@@ -122,6 +122,7 @@
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-war-plugin</artifactId>
+                <version>3.3.1</version>
```
Comment (piraveena) [inline components/org.wso2.carbon.identity.oauth.dcr.endpoint/pom.xml:125]: define the version in the parent pom file

File: components/org.wso2.carbon.identity.oauth.endpoint/pom.xml [hunk 2]
```diff
@@ -217,7 +227,7 @@
             </plugin>
             <plugin>
                 <artifactId>maven-war-plugin</artifactId>
-                <version>2.2</version>
+                <version>3.3.1</version>
```
Comment (piraveena) [inline components/org.wso2.carbon.identity.oauth.endpoint/pom.xml:230]: define this at the parent pom file under versions Eg: ${maven.war.plugin.version}. Fix this in all places

File: components/org.wso2.carbon.identity.discovery/src/test/java/org/wso2/carbon/identity/discovery/DefaultOIDCProcessorTest.java [hunk 3]
```diff
@@ -42,7 +43,7 @@
 /**
  * Unit test covering DefaultOIDCProcessor class.
  */
-public class DefaultOIDCProcessorTest {
+public class DefaultOIDCProcessorTest extends PowerMockTestCase {
```
[the remark this answers is not in the document]
Comment (YogyaGamage) [reply components/org.wso2.carbon.identity.discovery/src/test/java/org/wso2/carbon/identity/discovery/DefaultOIDCProcessorTest.java:46]: With the method _beforePowerMockTestClass()_ in PowerMockTestCase we can switch regular class loader to powermock class loader, without that it gives the error "_java.lang.IllegalStateException: Could not initialize plugin: interface org.mockito.plugins.MockMaker (alternate: null)_".

File: components/org.wso2.carbon.identity.oauth.client.authn.filter/spotbugs-exclude.xml [hunk 4]
```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (senthalan) [inline components/org.wso2.carbon.identity.oauth.client.authn.filter/spotbugs-exclude.xml:1]: include license header

**Pull request merged**